src/core/tsi/s2a/record_protocol/s2a_crypter_util.cc

```diff
@@ -20,33 +20,25 @@
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
+#include "src/core/tsi/s2a/s2a_constants.h"
 
-grpc_status_code s2a_deserialize_session_state(
```

Comment:
It is needed by the HS client, so a (slightly updated) version of this method is added back in in PR #21. (The next PR you're reviewing.)

It was originally placed in the record protocol util file when the `s2a_crypter_create` method took in a `grpc_byte_buffer` as input. Since that is changed, the record protocol no longer needs to deserialize anything.